advanced_functionality/distributed_tensorflow_mask_rcnn/mask-rcnn-s3.ipynb

```diff
@@ -0,0 +1,566 @@
+{
```

Comment:
<p>A common technique in SageMaker notebook examples is to use the "default bucket" provided by the SageMaker Python SDK. You would just need to create a <code>sagemaker.Session</code> instance first, which you're already doing in some later cells, so you could consider moving some of the SageMaker initialization steps to the top.</p><p> </p><p>I was able to do it minimally by adding one additional cell to initialize the session, then injecting the default bucket into the bash command.</p><p> </p><p><code>sage_sess = sagemaker.session.Session(boto_session=session)</code></p><p><code>bucket = sage_sess.default_bucket()</code></p><p>---</p><p><code>%%time</code></p><p><code># usage: ./prepare-s3-bucket.sh &lt;s3-bucket-name&gt;</code></p><p><code>!./prepare-s3-bucket.sh {bucket}</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/891/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Reply:
This common approach of implicitly creating an S3 bucket and using it in notebook is not applicable in this case. There is a setup cloud formation automation that accompanies this example that creates VPCs, VPC S3 endpoint, VPC S3 endpoint policy, SageMaker execution role to provide access to a specific S3 bucket. This use case is aimed at private VPC.

Reply:
OK, got it. It may still be helpful to have the user set Python variables `bucket` and `region` toward the top of the notebook, since they are required args to each of the shell scripts, and they need to be set later anyway to facilitate SageMaker steps.

Reply:
I am not opposed to this in principle. Can we defer it to next iteration as I want to avoid another round of testing and this change will definitely require another round of testing.

Reply:
Looks like we are pushing merge back to October 3 so I may be able to do this as well.

Reply:
Looks like you anyway did add the `region` and `s3_bucket` variables in recent revisions. I think this is helpful. Thanks!